presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorMetadata.java

```diff
@@ -925,6 +925,28 @@ default ConnectorTableProperties getTableProperties(ConnectorSession session, Co
         return Optional.empty();
     }
 
+    /**
+     * Attempt to push down the topN into the table scan.
+     * <p>
+     * Connectors can indicate whether they don't support topN pushdown or that the action had no effect
+     * by returning {@link Optional#empty()}. Connectors should expect this method may be called multiple times.
+     * </p>
+     * <b>Note</b>: it's critical for connectors to return {@link Optional#empty()} if calling this method has no effect for that
+     * invocation, even if the connector generally supports topN pushdown. Doing otherwise can cause the optimizer
+     * to loop indefinitely.
+     * <p>
+     * If the connector can handle TopN Pushdown it should return a new table handle which will replace the existing
+     * table handle in the TableScan.
```

Comment:
we do not expect a need to remap column handles, do we?

Reply:
I don't think that is needed as all the connector should need is a way to know what is the sort order and the limit and no column handle should change in any way.